packages/thunderstore-api/src/apiFetch.ts

```diff
@@ -39,32 +42,44 @@ async function fetchRetry(
   }
 }
 
+/**
+ * Simple timeout helper used by the retry loop.
+ */
 function sleep(delay: number) {
   return new Promise((resolve) => setTimeout(resolve, delay));
 }
 
-export type apiFetchArgs<B, QP> = {
+/**
+ * Arguments supplied to `apiFetch` describing the HTTP request and validation schemas.
+ */
+type SchemaOrUndefined<Schema extends z.ZodSchema | undefined> =
+  Schema extends z.ZodSchema ? z.infer<Schema> : undefined;
+
+export type apiFetchArgs<
```

Comment:
I've noticed in the few last PRs that especially types seems to be exported even when they're used only locally. I would rather not export stuff just in case. It's easy to add export later if needed, removing it is more effort if it's used somewhere outside already. And I think it also increases the odds of IDEs autoimporting stuff from these files directly, instead of the root of the package where the public interface of the package should be defined.